src/lib.rs

```diff
@@ -1,14 +1,48 @@
 use std::fmt;
 
-/// TODO...
+/// A trait that can be used for creating and converting between valid [numerical](https://www.w3.org/TR/2018/REC-css-color-3-20180619/#numerical) color values.
 pub trait Color {
-    // TODO...
+    /// Constructs a valid CSS color string for a given color type.
+    ///
+    /// # Examples
+    /// ```
+    /// use css_colors::{Color, RGB, RGBA};
+    ///
+    /// let salmon = RGB { r: 250, g: 128, b: 114 };
+    /// let opaque_salmon = RGBA { r: 250, g: 128, b: 114, a: 0.5 };
+    ///
+    /// assert_eq!(salmon.to_css(), "rgb(250, 128, 114)");
+    /// assert_eq!(opaque_salmon.to_css(), "rgba(250, 128, 114, 0.5)");
+    /// ```
     fn to_css(self) -> String;
 
-    // TODO...
+    /// Constructs a RGB struct.
+    /// Can be used to convert between different color types (e.g. RGBA to RGB).
+    /// When converting from a color with an alpha (`a`) value, by default the
+    /// new returned RGB will not retain that value.
+    ///
+    /// # Examples
+    /// ```
+    /// use css_colors::{Color, RGB, RGBA};
+    ///
+    /// let opaque_tomato = RGBA { r: 255, g: 99, b: 71, a: 0.5 };
+    ///
+    /// assert_eq!(opaque_tomato.to_rgb(), RGB { r: 255, g: 99, b: 71 });
     fn to_rgb(self) -> RGB;
 
-    // TODO...
+    /// Constructs a RGBA struct.
```

Comment:
"Convert self into an its RGBA representation. When converting from a representation that does not supports an alpha channel (e.g. RGB), it will be treated as fully opaque (i.e. an `a` value of `1.0`)"